Tests/SwiftTestingTestableCombinePublishersTests/SwiftTestingTestableCombinePublishersTests.swift

```diff
@@ -0,0 +1,525 @@
+            .waitForExpectations(timeout: 1)
+    }
+    
+    @Test func testNotFailureValueCompletionFail() async {
+        await withKnownIssue("Incorrect assertion should fail") {
+            await PassthroughSubject<Void, MockError>()
+                .expectNotFailure(MockError.someProblem)
+                .waitForExpectations(timeout: 1)
+        }
+        
+        await withKnownIssue("Incorrect assertion should fail") {
+            await Fail<Void, MockError>(error: MockError.someProblem)
+                .expectNotFailure(MockError.someProblem)
+                .waitForExpectations(timeout: 1)
+        }
+    }
+    
+    @Test func testFailureClosureCompletion() async {
+        await Fail<Void, MockError>(error: MockError.someProblem)
+            .expectFailure({ #expect($0 == .someProblem) })
+            .waitForExpectations(timeout: 1)
+    }
+    
+    @Test func testFailureClosureCompletionFail() async {
+        await withKnownIssue("Incorrect assertion should fail") {
+            await Fail<Void, MockError>(error: MockError.someProblem)
+                .expectFailure({ #expect($0 == .otherProblem) })
+                .waitForExpectations(timeout: 1)
+        }
+        
+        await withKnownIssue("Incorrect assertion should fail") {
+            await PassthroughSubject<Void, MockError>()
+                .expectFailure({ #expect($0 == .someProblem) })
+                .waitForExpectations(timeout: 1)
+        }
+    }
+    
+    @Test func testVoid() async {
+        await Just<Void>(Void())
+            .expectVoid()
+            .waitForExpectations(timeout: 1)
+    }
+    
+    @Test func testVoidFail() async {
+        await withKnownIssue("Incorrect assertion should fail") {
+            await Empty<Void, Never>(completeImmediately: true)
+                .expectVoid()
+                .waitForExpectations(timeout: 1)
+        }
+    }
+    
+    // MARK: - Misc Tests
+    
+    @Test func testMultipleExpectations() async {
+        await ["cool"]
+            .publisher
+            .expect("cool")
+            .expect({ #expect($0 == "cool") })
+            .expectSuccess()
+            .expectCompletion()
+            .expectCompletion({ #expect($0 == .finished) })
+            .waitForExpectations(timeout: 1)
+    }
+    
+    @Test func testMultipleFailures() async {
+        await withKnownIssue("Incorrect assertion should fail") {
+            await PassthroughSubject<String, MockError>()
+                .expect("cool")
+                .expect({ #expect($0 == "cool") })
+                .expectFailure()
+                .expectFailure({ #expect($0 == .someProblem) })
+                .expectFailure(.someProblem)
+                .expectSuccess()
+                .expectCompletion()
+                .expectCompletion({ #expect($0 == .finished) })
+                .waitForExpectations(timeout: 1)
+        }
+    }
+    
+    // TODO: Why is this supposed to fail?
```

Comment:
This test, while not a common-sense test, validates that expectations should be fulfilled in the order that they are expressed.